### DISCUSSION TITLE
How to select one of multiple overlapping highlights

### DISCUSSION THREAD

**Discussion by ryanwwest** [koreader/koreader#10480] [Q&A] [answered]
Sometimes, I want to highlight the same text multiple times (e.g. highlight an entire paragraph with 'lighten' style to pay attention to it, then highlight a few words inside the paragraph with 'underline' style with a note). This is supported in the `.sdr` Lua file and displays, but I can only select one of the two highlights after creating them. I didn't create an issue because maybe there's a solution to this? If there isn't, I can make one.

![image](https://github.com/koreader/koreader/assets/31378238/19919606-c6e7-461a-88d4-f73ceebe193e)

**Comment by hius07** [comment, marked as answer]
Tapping the overlap region shows the popup menu for the highlight made first (usually, sometimes can be the second - depending on the highlights configurations/overlaps).
Adding a note to or deleting a highlight made second can be done via the Bookmarks list.
Changing a style or boundaries for the second highlight is impossible currently.

Technically: we stop searching the highlight containing the touch coordinates after the first found entry.
Maybe it is worth to continue searching in all the highlights and show a popup menu with all highlights found.

`function ReaderHighlight:onTapPageSavedHighlight(ges) (for pdf)`
`function ReaderHighlight:onTapXPointerSavedHighlight(ges) (for cre)`

**Reply by ryanwwest** [reply]
I think a popup menu with all highlights found (only when there are more than one, of course) would be the most intuitive. While I could also envision tapping on an already-selected word to 'cycle' to selecting the next highlight in the stack that contains that word (some other software works in this way), I'm guessing that would conflict with some other funcitonality already in place and also be hard to distinguish which highlight is actively selected without additional clicks (to e.g. view highlight type).

I'll make an issue to track this. Thanks for the quick answer, @hius07.

**Reply by ryanwwest** [reply]
https://github.com/koreader/koreader/issues/10482
